include/nlohmann/detail/macro_scope.hpp

```diff
@@ -31,6 +31,14 @@
     #define JSON_HAS_CPP_14
 #endif
 
+#ifdef JSON_HAS_CPP_17
+    #if __has_include(<optional>)
+        #include <optional>
+    #elif __has_include(<experimental/optional>)
+        #include <experimental/optional>
```

Comment:
Here we will have `std::experimental::optional` instead of `std::optional`. Perhaps it would be better to just `#include <optional>`.

Reply:
I think now that C++17 support is a bit more stable, we can indeed drop the experimental headers.

Reply:
Fixed in if [[ "${CXXFLAGS}" != "" ]]; then export CXXFLAGS=${CXXFLAGS}; fi.